homeassistant/components/proxmoxve/const.py

```diff
@@ -0,0 +1,231 @@
+"""Constants for Proxmox integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+from enum import Enum
+from typing import Final
+
+from homeassistant.components.sensor import SensorStateClass
+from homeassistant.const import (
+    ATTR_TIME,
+    DATA_GIGABYTES,
+    DATA_RATE_MEGABYTES_PER_SECOND,
+    PERCENTAGE,
+)
+
+from .model import (
+    ProxmoxBinarySensorDescription,
+    ProxmoxButtonDescription,
+    ProxmoxSensorDescription,
+    ProxmoxSwitchDescription,
+)
+
+
+class Node_Type(Enum):
+    """Defines Node Types as VM or Containers."""
+
+    TYPE_VM = 0
+    TYPE_CONTAINER = 1
+
+
+DOMAIN = "proxmoxve"
+PROXMOX_CLIENTS = "proxmox_clients"
+COORDINATORS = "coordinators"
+
+CONF_REALM = "realm"
+CONF_NODE = "node"
+CONF_NODES = "nodes"
+CONF_VMS = "vms"
+CONF_CONTAINERS = "containers"
+
+DEFAULT_PORT = 8006
+DEFAULT_REALM = "pam"
+DEFAULT_VERIFY_SSL = True
+DEFAULT_SCAN_INTERVAL = 60
+
+
+COMMAND_REBOOT = "reboot"
+COMMAND_RESUME = "resume"
+COMMAND_SHUTDOWN = "shutdown"
+COMMAND_START = "start"
+COMMAND_STOP = "stop"
+COMMAND_SUSPEND = "suspend"  # API notes 'This is experimental.'  https://pve.proxmox.com/pve-docs/api-viewer/#/nodes/{node}/lxc/{vmid}/status/suspend
+
+VM_COMMANDS = (
+    COMMAND_REBOOT,
+    COMMAND_RESUME,
+    COMMAND_SHUTDOWN,
+    COMMAND_START,
+    COMMAND_STOP,
+    COMMAND_SUSPEND,
+)
+
+
+PROXMOX_BINARYSENSOR_TYPES: Final[tuple[ProxmoxBinarySensorDescription, ...]] = (
+    ProxmoxBinarySensorDescription(
+        key="status",
+        icon="mdi:server",
+    ),
+)
+
+
+PROXMOX_SENSOR_TYPES: Final[tuple[ProxmoxSensorDescription, ...]] = (
+    ProxmoxSensorDescription(
+        key="uptime",
+        icon="mdi:database-clock-outline",
+        unit_metric=ATTR_TIME,
+        unit_imperial=ATTR_TIME,
+        conversion=lambda x: timedelta(seconds=x),
```

Comment:
You can use a sensor with device class TIMESTAMP and let the frontend handle the time difference.

```suggestion
        conversion=lambda x: ( dt_util.utcnow()-timedelta(seconds=x)),
        device_class=SensorDeviceClass.TIMESTAMP,
```

Remembering to include the necessary imports:
````
import homeassistant.util.dt as dt_util
````
and 
````
from homeassistant.components.sensor import SensorDeviceClass
````